### PR TITLE
docs: extended locking documentation

### PR DESCRIPTION
**Pull request by SimonTheSourcerer**
### Description
Extended locking documentation which additional information about how to integrated locking in your project.

### CODE REVIEW

File: .Documentation/articles/developer/10-lockable-properties.md [hunk 1]
```diff
@@ -43,8 +43,32 @@ public class LightSwitchProperty : LockableProperty
 }
 ```
 
-## Disabling The Locking Feature
+## Step Lock Handling
+The logic of locking and unlocking LockableProperties is provided by an abstract class `StepLockHandlingStrategy`. Which cares about locking LockableProperties per step. It also allows to be configured by the active mode. The default strategy is named `DefaultStepLockHandling`. To deactivate the locking feature, set the `StepLockHandling` property of your runtime configuration to an instance of the `NonLockingStepHandling` class.
```
Comment (tomwim) [inline .Documentation/articles/developer/10-lockable-properties.md:47]:
```suggestion
The logic of locking and unlocking LockableProperties is provided by an abstract class `StepLockHandlingStrategy`, which cares about locking LockableProperties per step. It also allows to be configured by the active mode. The default strategy is named `DefaultStepLockHandling`. To deactivate the locking feature, set the `StepLockHandling` property of your runtime configuration to an instance of the `NonLockingStepHandling` class.
```

File: .Documentation/articles/developer/10-lockable-properties.md [hunk 2]
```diff
@@ -43,8 +43,32 @@ public class LightSwitchProperty : LockableProperty
...
 
-To disable the locking feature, set the `StepLockHandling` property of your runtime configuration to an instance of the `Innoactive.Creator.Core.RestrictiveEnvironment.NonLockingStepHandling` class.
+### DefaultStepLockHandling
+Locks & unlocks LockableProperties according to the usage in the step. It also looks up which of the properties are used in the following step and keeps them unlocked. Additionally it allows to be configured by the current active Mode, following Parameter are allowed:
```
Comment (tomwim) [inline .Documentation/articles/developer/10-lockable-properties.md:50]:
```suggestion
Locks & unlocks LockableProperties according to the usage in the step. It also looks up which of the properties are used in the following step and keeps them unlocked. Additionally it allows to be configured by the current active Mode, following parameters are allowed:
```

File: .Documentation/articles/developer/10-lockable-properties.md [hunk 3]
```diff
@@ -43,8 +43,32 @@ public class LightSwitchProperty : LockableProperty
...
+
+Name | type | default value | comment
+|---|---|---|---|
+LockOnCourseStart | `boolean` | `true` | *Decides if all LockableProperties should get locked when the course is started*
+LockOnCourseFinished | `boolean` | `false` | *Decides if all LockableProperties should get locked when the course is finished*
+
+## Extending Conditions
+All Conditions based on the abstract class `Condition` have an already existing implementation of the method provided by the interface `ILockablePropertiesProvider`, which extract all `LockableProperty` from the Condition's data. If you want to manually intervene and change the value or the specific properties used, you are able to override the method.
```
Comment (tomwim) [inline .Documentation/articles/developer/10-lockable-properties.md:58]:
```suggestion
All Conditions based on the abstract class `Condition` have an already existing implementation of the method provided by the interface `ILockablePropertiesProvider`, which extracts all `LockableProperty` from the Condition's data. If you want to manually intervene and change the value or the specific properties used, you are able to override the method.
```

File: .Documentation/articles/developer/10-lockable-properties.md [hunk 4]
```diff
@@ -43,8 +43,32 @@ public class LightSwitchProperty : LockableProperty
...
+
+An example in `GrabbedCondition` which keeps the properties unlocked after finishing the Step:
```
Comment (tomwim) [inline .Documentation/articles/developer/10-lockable-properties.md:60]:
```suggestion
An example in `GrabbedCondition` which keeps the properties unlocked after finishing the step:
```

File: .Documentation/articles/developer/10-lockable-properties.md (hunk 1 above)
Comment (Luphoxa) [reply .Documentation/articles/developer/10-lockable-properties.md:47]: I would also suggest:
```suggestion
The logic of locking and unlocking LockableProperties is provided by the abstract class `StepLockHandlingStrategy`, which cares about locking LockableProperties per step. It also allows being configured by the active mode. The default strategy is named `DefaultStepLockHandling`. To deactivate the locking feature, set the `StepLockHandling` property of your runtime configuration to an instance of the `NonLockingStepHandling` class.
```

File: .Documentation/articles/developer/10-lockable-properties.md [hunk 5]
```diff
@@ -43,8 +43,32 @@ public class LightSwitchProperty : LockableProperty
...
 
-To disable the locking feature, set the `StepLockHandling` property of your runtime configuration to an instance of the `Innoactive.Creator.Core.RestrictiveEnvironment.NonLockingStepHandling` class.
+### DefaultStepLockHandling
+Locks & unlocks LockableProperties according to the usage in the step. It also looks up which of the properties are used in the following step and keeps them unlocked. Additionally it allows to be configured by the current active Mode, following parameters are allowed:
+
+Name | type | default value | comment
+|---|---|---|---|
+LockOnCourseStart | `boolean` | `true` | *Decides if all LockableProperties should get locked when the course is started*
+LockOnCourseFinished | `boolean` | `false` | *Decides if all LockableProperties should get locked when the course is finished*
```
Comment (Maxsiom) [inline .Documentation/articles/developer/10-lockable-properties.md:55]: `EnableLockHandling` missing?

Comment (Luphoxa) [reply .Documentation/articles/developer/10-lockable-properties.md:55]: In the list or in general?

Comment (tomwim) [reply .Documentation/articles/developer/10-lockable-properties.md]: good catch, added it

Comment (tomwim) [reply .Documentation/articles/developer/10-lockable-properties.md]: I think this includes 117

**Pull request merged**